homeassistant/components/energyzero/__init__.py

```diff
@@ -1,17 +1,56 @@
 """The EnergyZero integration."""
 from __future__ import annotations
 
+from energyzero import EnergyZero
+
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import Platform
-from homeassistant.core import HomeAssistant
+from homeassistant.core import (
+    HomeAssistant,
+    ServiceCall,
+    ServiceResponse,
+    SupportsResponse,
+)
 from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.util import dt as dt_util
 
-from .const import DOMAIN
+from .const import DOMAIN, SERVICE_NAME, SERVICE_PRICE_TYPES, SERVICE_SCHEMA
 from .coordinator import EnergyZeroDataUpdateCoordinator
 
 PLATFORMS = [Platform.SENSOR]
 
 
+async def _get_prices(hass: HomeAssistant, call: ServiceCall) -> ServiceResponse:
+    """Search prices."""
+    price_type = call.data["type"]
+
+    if price_type not in SERVICE_PRICE_TYPES:
+        raise ValueError(f"Invalid service type: {call.data['type']}")
+
+    energyzero = EnergyZero(
+        session=async_get_clientsession(hass),
+        incl_btw="true" if call.data["incl_btw"] else "false",
+    )
+
+    start = dt_util.parse_datetime(call.data.get("start") or "") or dt_util.now().date()
```

Comment:
```suggestion
    start = dt_util.now().date()
    if start_field := call.data.get("start", ""):
        if value := dt_util.parse_datetime(start_field):
            start = value
        # todo raise error
```
We should raise when a user inserted date is invalid